src/features/import/components/Upload/UploadFile.tsx

```diff
@@ -0,0 +1,177 @@
+    },
+    maxFiles: 1,
+    noClick: true,
+    noKeyboard: true,
+    onDrop,
+    onDropAccepted: () => setError(false),
+    onDropRejected: () => setError(true),
+    onError: () => {
+      setError(true);
+    },
+  });
+
+  return (
+    <>
+      <Box
+        {...getRootProps()}
+        p={12}
+        style={
+          loading
+            ? {
+                alignItems: 'center',
+                backgroundColor: '#1976D214',
+                borderColor: '#1976D2',
+                borderRadius: 4,
+                borderStyle: 'dashed',
+                borderWidth: 2,
+                display: 'flex',
+                flexDirection: 'column',
+                flexWrap: 'wrap',
+                margin: 16,
+                textAlign: 'center',
+              }
+            : {
+                alignItems: 'center',
+                backgroundColor: error ? '#fdf7f7' : 'transparent',
+                borderColor: error ? theme.palette.error.dark : '#E0E0E0',
+                borderRadius: 4,
+                borderStyle: error ? 'solid' : 'dashed',
+                borderWidth: 2,
+                display: 'flex',
+                flexDirection: 'column',
+                flexWrap: 'wrap',
+                margin: 16,
+                textAlign: 'center',
+              }
+        }
+      >
+        {loading && (
+          <Box>
+            <CircularProgress sx={{ color: '#1976D2' }} />
+            <Typography>{messages.uploadDialog.loading()}</Typography>
+          </Box>
+        )}
+        <input type="file" {...getInputProps()} />
+        <Box>
+          {!loading && (
+            <Box>
+              <IconButton
+                onClick={open}
+                style={{
+                  backgroundColor: '#E0E0E0',
+                  borderRadius: 100,
+                  cursor: 'pointer',
+                  height: 40,
+                  padding: 30,
+                  width: 40,
+                }}
+                type="button"
+              >
+                <UploadFileOutlined
+                  sx={{ color: theme.palette.primary.main, fontSize: 40 }}
+                />
+              </IconButton>
+              {isDragActive ? (
+                <Typography>{messages.uploadDialog.release()}</Typography>
+              ) : (
+                <>
+                  <Box pt={2}>
+                    <Typography
+                      onClick={open}
+                      sx={{
+                        color: theme.palette.primary.main,
+                        cursor: 'pointer',
+                        display: 'inline-block',
+                        padding: 1,
+                        textDecorationLine: 'underline',
+                      }}
+                    >
+                      {messages.uploadDialog.instructions()}
+                    </Typography>
+                    <Typography
+                      sx={{
+                        display: 'inline-block',
+                      }}
```

Comment:
I would generally advice against this. What's being created here is a `p` tag (I think, that's the default for `Typography`, right?) that is then rendered as `inline-block`. It looks good, but semantically it's two separate paragraphs, even if the latter is rendered "inline" with the former.

I believe there is a `component` property for `Typography` where you can pass "span" if you want to render something as a span of text within another text (or in whatever context makes you prefer `span`).